### ISSUE TITLE
Ensure that each VM has private SSH and TLS keys

### ISSUE THREAD

**Issue by DemiMarie** [QubesOS/qubes-issues#7685]
[How to file a helpful issue](https://www.qubes-os.org/doc/issue-tracking/)

### The problem you're addressing (if any)

TemplateBasedVMs use the same SSH and TLS private keys as their TemplateVMs by default

### The solution you'd like

Each Qubes VM should have SSH and TLS keys that are specific to that VM, and not available in other VMs.

### The value to a user, and who that user might be

All users who manage Qubes OS servers will benefit from easier management and improved security.

**Comment by ben-grande** [comment]
Hm... TLS I don't know, doesn't each webserver create their own certs?

For SSH, keys could be deleted on boot (dirty) or be excluded somehow from ever being shared from the root volume, I don't think the current templating system supports this.

**Comment by DemiMarie** [comment]
What about disabling SSH key generation in templates, and bind-mounting the relevant directories outside of them?

**Comment by ben-grande** [comment]
Yes, it is a better idea. I don't know if it is possible to disable key generation if it is done by the package manager, during post install:

https://salsa.debian.org/ssh-team/openssh/-/blob/master/debian/openssh-server.postinst?ref_type=heads#L27-L57

Although we could add something similar to `ExecStartPre` of the ssh service, I guess for template, we need to wait for the key to be generated and a package manager hook deletes the keys if it is a template.
